pages/search.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from "react";
+import SearchForm from "../components/SearchForm";
+import axios from "../lib/axios";
+import { useRouter } from "next/router";
+
+interface Article {
+  content: string;
+  createdAt: string;
+  id: number;
+  image: string | null;
+  likeCount: number;
+  title: string;
+  writer: { id: number; nickname: string };
+}
+export default function Search() {
+  const [articles, setArticles] = useState<Article[]>([]);
+  const router = useRouter();
+  const {q} = router.query;
+
+  async function getArticles(keyword: string) {
+    try {
+      const res = await axios.get(`/articles?keyword=${keyword}`);
+      const nextArticles = res.data.results ?? [];
+      console.log(nextArticles);
+      setArticles(nextArticles);
+    } catch (error) {
+      console.error("Error fetching articles:", error);
+    }
+  }
+  useEffect(() => {
+    if (q) {
+      getArticles(q as string);
+      console.log(q);
+    }
+  }, [q]);
+
+  return (
+    <>
+      <h2>{q} 결과 </h2>
+      <SearchForm initialValue={q as string} />
+      {articles && articles.length > 0 ? (
+        articles.map((article) => (
+          <div key={article.id}>
+            <h2>{article.title}</h2>
+            <p>{article.content}</p>
+            <p>By: {article.writer.nickname}</p>
+            <p>Likes: {article.likeCount}</p>
+          </div>
+        ))
```

Comment:
아래와 같이 구조분해 할당을 사용하면 구현부를 조금 더 간단하게 작성할 수 있습니다!

```suggestion
        articles.map(({id, title, content, writer: { nickname }, likeCount }) => (
          <div key={id}>
            <h2>{title}</h2>
            <p>{content}</p>
            <p>By: {nickname}</p>
            <p>Likes: {likeCount}</p>
          </div>
        ))
```